lib/cdo/delete_accounts_helper.rb

```diff
@@ -316,7 +332,8 @@ def purge_user(user)
 
     # Cache user email here before destroying user; migrated users have their
     # emails stored in primary_contact_info, which will be destroyed.
-    user_email = user.email
+    # If the user account was already soft-deleted, then fallback to the :email attribute.
+    user_email = (user.email&.blank?) ? user.read_attribute(:email) : user.email
```

Comment:
This is because `.email` is actually a method, not a field. If the account is "migrated", then `.email` will defer to the `primary_contact_info`. However, if the account is soft-deleted, then calling `user.email` won't find the `primary_contact_info` because that will have been soft-deleted too.

Here is the code for the email method.
```
def email
    return read_attribute(:email) unless migrated?
    primary_contact_info.try(:email) || ''
end
```

Rather than update `email` to handle looking up the primary_contact_info if the account is soft-deleted, I instead updated the `delete_accounts_helper.rb` to handle looking up the email for soft-deleted accounts.

Let me know if you think I could have done this a better way or if this explanation wasn't clear.

Reply:
That makes sense! I figured there was a piece I missed and this explains it. 🙂